cli/tests/environment-list.bats

```diff
@@ -122,3 +122,17 @@ EOF
   assert_success
   assert_output "hello"
 }
+
+# ---------------------------------------------------------------------------- #
+
+# https://github.com/flox/flox/issues/1039
+# bats test_tags=list,list:tolerates-missing-version
+@test "'flox list' tolerates missing version" {
+  "$FLOX_BIN" init
+  # `influxdb does not have a version attribute set in nixpkgs (2024-02-19)
+  # todo: replace with a more predicatable/smaller example
+  "$FLOX_BIN" install influxdb2
+  run "$FLOX_BIN" list
+  assert_success
+  assert_output "influxdb2: influxdb2 (N/A)"
+}
```

Comment:
nonblocking: can we add this as a unit test? I think we need to be a bit more diligent about making things unit tests when we can for the sake of test suite time

Reply:
Unless we want to test the output as well I can make this a unit test instead, yes

Reply:
I think the unit test is probably sufficient